server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentHelper.java

```diff
@@ -74,6 +74,20 @@ public static Iterator<ToXContent> field(String name, String value) {
         return Iterators.single(((builder, params) -> builder.field(name, value)));
     }
 
+    /**
+     * Creates an Iterator to serialize a named field where the value is represented by a chunked ToXContext.
+     * Chunked equivalent for {@code XContentBuilder field(String name, ToXContent value)}
+     * @param name name of the field
+     * @param value ChunkedToXContent value for this field (single value, object or array)
+     * @return Iterator composing field name and value serialization
+     */
+    public static Iterator<ToXContent> field(String name, ChunkedToXContent value) {
```

Comment:
I think this has to be a full object, not a fragment:

```suggestion
    public static Iterator<ToXContent> field(String name, ChunkedToXContentObject value) {
```

Reply:
Yes, good one, I tried to "enforce" this via the javadoc comment but using the more specific type is much better